vcd/resource_vcd_vapp_vm.go

```diff
@@ -2031,3 +2112,175 @@ func setGuestCustomizationData(d *schema.ResourceData, vm *govcd.VM) error {
 
 	return nil
 }
+
+func addEmptyVm(d *schema.ResourceData, vcdClient *VCDClient, org *govcd.Org, vdc *govcd.Vdc, vapp *govcd.VApp) (*govcd.VM, error) {
+	log.Printf("[TRACE] Creating empty VM: %s", d.Get("name").(string))
+
+	var ok bool
+	var memory interface{}
+	if memory, ok = d.GetOk("memory"); !ok {
+		return nil, fmt.Errorf("`memory` is required when creating empty VM")
+	}
+
+	var osType interface{}
+	if osType, ok = d.GetOk("os_type"); !ok {
+		return nil, fmt.Errorf("`os_type` is required when creating empty VM")
+	}
+
+	if _, ok = d.GetOk("network_name"); ok {
+		return nil, fmt.Errorf("`network_name` can't be used with empty VM, please use `network` configuration")
+	}
+
+	if _, ok = d.GetOk("vapp_network_name"); ok {
+		return nil, fmt.Errorf("`vapp_network_name` can't be used with empty VM, please use `network` configuration")
+	}
+
+	var hardWareVersion interface{}
+	if hardWareVersion, ok = d.GetOk("hardware_version"); !ok {
+		return nil, fmt.Errorf("`hardware_version` is required when creating empty VM")
+	}
+
+	var computerName interface{}
+	if computerName, ok = d.GetOk("computer_name"); !ok {
+		return nil, fmt.Errorf("`computer_name` is required when creating empty VM")
+	}
+
+	var bootImage *types.Media
+	if bootImageName, ok := d.GetOk("boot_image"); ok {
+		result, err := vdc.QueryAllMedia(bootImageName.(string))
+		if err != nil || len(result) == 0 {
+			return nil, fmt.Errorf("[VM creation] error getting boot image %s : %s", bootImageName, err)
+		}
+		if len(result) > 1 {
+			return nil, fmt.Errorf("[VM creation] error found more than one boot image %s : %s", bootImageName, err)
```

Comment:
the error is `nil` and should not be shown here.

Reply:
fixed